flink-ml-lib/src/test/java/org/apache/flink/ml/classification/OnlineModelSaveLoadTest.java

```diff
@@ -0,0 +1,206 @@
+                            Row.of(Vectors.dense(3.1, 4.6, 3.1)),
+                            Row.of(Vectors.dense(20.1, 5.6, 3.1)),
+                            Row.of(Vectors.dense(2.1, 4.7, 3.1))));
+    private String tmpPath;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        env.setParallelism(1);
+        tEnv = StreamTableEnvironment.create(env);
+    }
+
+    private void saveModelData() {
+        Schema modelSchema =
+                Schema.newBuilder().column("f0", DataTypes.of(DenseVector.class)).build();
+        Table modelDataTable = tEnv.fromDataStream(env.fromCollection(modelRows), modelSchema);
+        try {
+            tmpPath = tempFolder.newFolder().getAbsolutePath();
+            /* Saves online model data to given path (tmpPath). */
+            saveModelData(
+                    LogisticRegressionModelData.getModelDataStream(modelDataTable),
+                    tmpPath,
+                    new LogisticRegressionModelData.ModelDataEncoder());
+            env.execute();
+        } catch (Exception e) {
+            throw new RuntimeException("Saves model data failed!");
+        }
+    }
+
+    @Test
+    public void testSaveAndLoadSingleModel() throws Exception {
+        saveModelData();
+        /* Loads model with modelVersion equals "2" in model path. */
+        String modelVersion = "2";
+        Table modelData =
+                tEnv.fromDataStream(
+                        loadModelData(
+                                env,
+                                tmpPath,
+                                new LogisticRegressionModelData.ModelDataDecoder(),
+                                modelVersion));
+        List<Row> loadedModelRows =
+                IteratorUtils.toList(tEnv.toDataStream(modelData).executeAndCollect());
+        Assert.assertEquals(modelRows.get(2), loadedModelRows.get(0));
+    }
+
+    @Test
+    public void testSaveAndLoadOnlineModel() throws Exception {
+        saveModelData();
+        Table modelData =
+                tEnv.fromDataStream(
+                        ReadWriteUtils.loadModelData(
+                                env, tmpPath, new LogisticRegressionModelData.ModelDataDecoder()));
+
+        List<Row> loadedModelRows =
+                IteratorUtils.toList(tEnv.toDataStream(modelData).executeAndCollect());
+
+        TestBaseUtils.compareResultCollections(
+                loadedModelRows,
+                modelRows,
+                (o1, o2) -> {
+                    DenseVector v1 = (DenseVector) o1.getField(0);
+                    DenseVector v2 = (DenseVector) o2.getField(0);
+
+                    if (v1.values[0] > v2.values[0]) {
```

Comment:
How about replace the code snippet with something like this:

`
for (int x = 0; x < min(first.length, second.length); x++) {
            int cmp = Double.compare(first[x], second[x]);
            if (cmp != 0) {
                return cmp;
            }
        }
        int cmp = first.length - second.length;
        return cmp;
`